### ISSUE TITLE
Do not test ETAG values

### ISSUE THREAD

**Issue by laurentsenta** [ipfs/gateway-conformance#73]
Brought up by @lidel and @hacdias on Slack:

https://filecoinproject.slack.com/archives/C04M8232QRW/p1685573605787349?thread_ts=1685531506.240809&cid=C04M8232QRW

> For the Etags specifically , the best IMO would be to move those tests to Boxo and remove them from both Kubo and the conformance. I think that’s something we should agree on.

> yeah, conformance is expecting boxo/kubo-specific Etag values ([example](https://github.com/ipfs/gateway-conformance/blob/e813393211643a905d3a62e7c547bc6c1e457028/tests/t0116_gateway_cache_test.go#L35)) because MVP was to port sharness.
> next step is to make these tests vendor-agnostic:
> Conformance should not care what is inside of Etag, but must test Etag is present, and unique to specific resource+response type.
Boxo/Kubo can test specific values.

**Comment by hacdias** [comment]
Yeah, I already changed the CAR ones in #56 to only check for its presence, but not the value itself.

**Comment by hacdias** [comment]
We should also note that tests with `If-None-Match` would not work properly, because they assume Kubo Etags. We should go over this tests and see which ones are Boxo specific. I think Etag and If-None-Match testing with specific values should be directly done as unit tests in Boxo.

**Comment by galargh** [comment]
> I'm not sure if this is something we can do right now: use the values of one request for the next. cc @laurentsenta @galargh

Not yet but it's something we've been thinking about for a while and waiting for when the actual need arises - it seems it just has. We've been playing around with this idea in https://github.com/ipfs/gateway-conformance/pull/30. The PR itself is not longer relevant in its' current form but there are some parts of it that we'll salvage.
